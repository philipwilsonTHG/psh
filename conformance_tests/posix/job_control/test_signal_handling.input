# Test signal handling with trap builtin
# PSH supports trap builtin for signal handling (v0.57.3 feature)

# Test trap listing
echo "Testing trap -l (list signals):"
trap -l | head -5

# Test trap display
echo "Testing trap -p (show current traps):"
trap -p

# Test setting a trap for EXIT
echo "Testing EXIT trap:"
trap 'echo "EXIT trap executed"' EXIT

# Test function in trap
cleanup_func() {
    echo "Cleanup function called"
}

trap cleanup_func EXIT

# Test trap with signal names
echo "Testing trap with signal names:"
trap 'echo "TERM signal trapped"' TERM

# Show the TERM trap
trap -p TERM

# Test trap reset
echo "Testing trap reset:"
trap - TERM
trap -p TERM

# Test trap with signal numbers
echo "Testing trap with signal numbers:"
trap 'echo "Signal 15 trapped"' 15
trap -p 15

# Test trap ignore (empty string)
echo "Testing trap ignore:"
trap '' HUP
trap -p HUP

# Test multiple signals
echo "Testing multiple signals:"
trap 'echo "Multiple signal trap"' INT QUIT
trap -p INT
trap -p QUIT

# Test trap with quotes
echo "Testing trap with complex action:"
trap 'echo "Complex trap: PID=$$, Time=$(date)"' USR1
trap -p USR1

# Reset all traps for clean exit
trap - INT QUIT USR1 HUP 15

echo "Signal handling test completed"

# Note: The EXIT trap will fire when the script exits