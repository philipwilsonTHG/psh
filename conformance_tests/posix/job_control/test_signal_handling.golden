Testing trap -l (list signals):
Testing trap -p (show current traps):
Testing EXIT trap:
Testing trap with signal names:
trap -- 'echo "TERM signal trapped"' TERM
Testing trap reset:
Testing trap with signal numbers:
trap -- 'echo "Signal 15 trapped"' 15
Testing trap ignore:
trap -- '' HUP
Testing multiple signals:
trap -- 'echo "Multiple signal trap"' INT
trap -- 'echo "Multiple signal trap"' QUIT
Testing trap with complex action:
trap -- 'echo "Complex trap: PID=$$, Time=$(date)"' USR1
Signal handling test completed
